packages/@react-stately/data/src/useAsyncList.ts

```diff
@@ -151,12 +152,13 @@ function reducer<T, C>(data: AsyncListState<T, C>, action: Action<T, C>): AsyncL
             return data;
           }
 
+          selectedKeys = action.selectedKeys ?? data.selectedKeys;
           return {
             ...data,
             filterText: action.filterText ?? data.filterText,
             state: 'idle',
             items: [...action.items],
-            selectedKeys: new Set(action.selectedKeys ?? data.selectedKeys),
+            selectedKeys: selectedKeys === 'all' ? 'all' : new Set(selectedKeys),
```

Comment:
I'm surprised there isn't a type update to this? https://github.com/adobe/react-spectrum/blob/main/packages/@react-stately/data/src/useAsyncList.ts#L70
any idea why?

Reply:
I don't think there needs to be a type update since `Selection` takes `'all' | Set<Key>`